commands/orchestrationcommands/stackcommands/adopt_test.go

```diff
@@ -36,3 +40,57 @@ func TestAdoptServiceClientType(t *testing.T) {
 	actual := cmd.ServiceClientType()
 	th.AssertEquals(t, expected, actual)
 }
+
+func TestAdoptPreCSV(t *testing.T) {
+	cmd := &commandAbandon{
+		Ctx: &handler.Context{
+			ServiceClient: client.ServiceClient(),
+		},
+	}
+
+	resource := &handler.Resource{
+		Params: &paramsAdopt{},
+	}
+
+	expected := "{\"ID\":\"16ef0584-4458-41eb-87c8-0dc8d5f66c87\",\"Links\":[{\"Href\":\"http://168.28.170.117:8004/v1/98606384f58drad0bhdb7d02779549ac/stacks/stackcreated/16ef0584-4458-41eb-87c8-0dc8d5f66c87\",\"Rel\":\"self\"}]}"
```

Comment:
Ah, good question. What is happening is that the `FlattenMap` method does not handle the keys correctly if they're `gophercloud.Link` type. In the code, the method `stackSingle` (or something similar) is called on the result to convert the `Links` to a `map[string]interface{}` or similar type.